### PR TITLE
fixes #298, fixes #297

### LINKED ISSUE #297: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIXPCComponents_Utils.import]
@undefined:43:undefined\n@resource://guide-cfr-1-at-shield-dot-mozilla-dot-org/lib/controller.js:43:18\nexports.main/<@resource://guide-cfr-1-at-shield-dot-mozilla-dot-org/lib/index.js:52:12\nHandler.prototype.process@resource://gre/modules/Promise-backend.js:934:23\nthis.PromiseWalker.walkerLoop@resource://gre/modules/Promise-backend.js:813:7\nthis.PromiseWalker.scheduleWalkerLoop/<@resource://gre/modules/Promise-backend.js:747:1

### LINKED ISSUE #298: loggerData undefined if silence period ends while Firefox is not open
To reproduce:

1- Put the system in silence period
2- Close Firefox
3- Wait for silence period to end
4- Start Firefox

The error is non-blocking and gets fixed the next time Firefox is opened.

### PR DESCRIPTION
**Pull request by raymak**

### CODE REVIEW

File: phase2/addon/lib/controller.js [hunk 1]
```diff
@@ -40,7 +40,14 @@ const {Task} = Cu.import("resource://gre/modules/Task.jsm");
 const {Services} = Cu.import("resource://gre/modules/Services.jsm");
 const {AddonManager} = Cu.import("resource://gre/modules/AddonManager.jsm");
 const {XPCOMUtils} = Cu.import("resource://gre/modules/XPCOMUtils.jsm");
-const devtools = Cu.import("resource://devtools/shared/Loader.jsm").devtools;
+
+let devtools;
+try {
```
Comment (raymak) [inline phase2/addon/lib/controller.js]: Tries to load devtools the new way. If it fails, tries the deprecated way.

File: phase2/addon/lib/timer.js [hunk 2]
```diff
@@ -49,8 +49,6 @@ const _init = function(){
   if (!("silenceStart" in timerData))
     timerData.silenceStart = -1;
 
-  silenceLeft(); //to update the silence status;
```
Comment (raymak) [inline phase2/addon/lib/timer.js]: Silence status is not checked when timer is initializing. It's checked after 1 minute or when a recommendation is being delivered.

**Comment by gregglind** [comment]
R+ assuming build works.

**Comment by gregglind** [comment]
(Tests seem to be failing, maybe build locally and check?)

**Comment by raymak** [comment]
Yes, I'm trying to figure out why they fail. It builds fine on my Mac. I'm trying on Linux now.

**Comment by raymak** [comment]
Tests are failing due to the new Firefox update (48) that forces addon signing.

**Pull request merged**